messenger/message-recorder.rst

```diff
@@ -0,0 +1,188 @@
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <framework:config>
+                <framework:messenger default_bus="messenger.bus.command">
+                  <framework:bus name="messenger.bus.command">
+                      <framework:middleware id="validation">
+                      <framework:middleware id="doctrine_transaction">
+                  </framework:bus>
+                  <framework:bus name="messenger.bus.command" default_middleware="allow_no_handlers">
+                      <framework:middleware id="validation">
+                      <framework:middleware id="doctrine_transaction">
+                  </framework:bus>
+                </framework:messenger>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/cache.php
+        $container->loadFromExtension('framework', [
+            'messenger' => [
+                'default_bus' => 'messenger.bus.command',
+                'buses' => [
+                    'messenger.bus.command' => [
+                        'middleware' => ['validation', 'doctrine_transaction'],
+                    ],
+                    'messenger.bus.event' => [
+                        'default_middleware' => 'allow_no_handlers',
+                        'middleware' => ['validation', 'doctrine_transaction'],
+                    ],
+                ],
+            ],
+        ]);
+
+.. code-block:: php
+
+    namespace App\Messenger\CommandHandler;
+
+    use App\Entity\User;
+    use App\Messenger\Command\SignUpUser;
+    use App\Messenger\Event\UserSignedUp;
+    use Doctrine\ORM\EntityManagerInterface;
+    use Symfony\Component\Messenger\Envelope;
+    use Symfony\Component\Messenger\Stamp\DispatchAfterCurrentBusStamp;
+    use Symfony\Component\Messenger\MessageBusInterface;
+
+    class SignUpUserHandler
+    {
+        private $em;
+        private $eventBus;
+
+        public function __construct(MessageBusInterface $eventBus, EntityManagerInterface $em)
+        {
+            $this->eventBus = $eventBus;
+            $this->em = $em;
+        }
+
+        public function __invoke(SignUpUser $command)
+        {
+            $user = new User($command->getUuid(), $command->getName(), $command->getEmail());
+            $this->em->persist($user);
+
+            // The DispatchAfterCurrentBusStamp marks the event message to be handled
+            // only if this handler does not throw an exception.
+
+            $event = new UserSignedUp($command->getUuid());
+            $this->eventBus->dispatch(
+                (new Envelope($event))
+                    ->with(new DispatchAfterCurrentBusStamp())
+            );
+        }
+    }
+
+.. code-block:: php
+
+    namespace App\Messenger\EventSubscriber;
+
+    use App\Entity\User;
+    use App\Messenger\Event\UserSignedUp;
+    use Doctrine\ORM\EntityManagerInterface;
```

Comment:
`MyMailer` use statement missing. Maybe you could use symfony/mailer ?

Reply:
Thanks!